pkg/operator/operator.go

```diff
@@ -191,6 +191,10 @@ func (o *Operator) DeployApp(appID string, sequence int64) (deployed bool, deplo
 		if err != nil {
 			logger.Error(errors.Wrap(err, "failed to update downstream status"))
 		}
+
+		if err := o.ensureKotsadmApplicationMetadataConfigMap(appID, util.PodNamespace); err != nil {
```

Comment:
this should be at the beginning if the defer, because there are return statements above and it might not get executed